### PR TITLE
Services passed as constructor dependencies

### PR DESCRIPTION
**Pull request by garf**
This is preparation for Symfony 4. As services will be private by default and using them from container is not optimal.

### CODE REVIEW

**Comment by dmaicher** [comment]
Thanks but actually this is done on purpose :wink: Its simply more performant to load the services lazy directly from the controller only if needed.

Similar issue: https://github.com/lexik/LexikCurrencyBundle/issues/18

So to prepare for Symfony 4 we should mark the services as public instead :blush:

File: src/DAMA/MenuBundle/Resources/config/services.yml [hunk 1]
```diff
@@ -1,7 +1,10 @@
 services:
   dama_menu.twig.menu_extension:
     class: DAMA\MenuBundle\Twig\MenuExtension
-    arguments: ["@service_container"]
+    arguments:
+      - "@twig"
```
Comment (dmaicher) [inline src/DAMA/MenuBundle/Resources/config/services.yml]: also wondering: does this not produce a circular dependency somehow?

Comment (garf) [reply src/DAMA/MenuBundle/Resources/config/services.yml]: Well, it works :D I guess not :)

Comment (dmaicher) [reply src/DAMA/MenuBundle/Resources/config/services.yml]: :smile:

**Comment by garf** [comment]
So I will change this PR later then :)

File: src/DAMA/MenuBundle/Resources/config/services.yml (hunk 1 above)
Comment (garf) [reply src/DAMA/MenuBundle/Resources/config/services.yml]: Just curious if pulling twig from container is not the same? :)

Comment (dmaicher) [reply src/DAMA/MenuBundle/Resources/config/services.yml]: not sure I follow? the `twig` service must already be instantiated anyway if we we using any of the methods provided by `DAMA\MenuBundle\Twig\MenuExtension` as we are rendering a template then :blush:

Comment (garf) [reply src/DAMA/MenuBundle/Resources/config/services.yml]: I mean is there a difference in pulling twig from injected container or inject a twig? Aren't they instantiated only once?

Comment (dmaicher) [reply src/DAMA/MenuBundle/Resources/config/services.yml]: Ah thats what you mean. For `@twig` it does not make any difference.

But for `@dama_menu.menu_factory` and `@dama_menu.menu_config_provider` it does make a difference as the methods of this extension might not be needed for a lot of requests so often we instantiate them for no reason.

Better be lazy here :blush:

Comment (garf) [reply src/DAMA/MenuBundle/Resources/config/services.yml]: I always thought that Public means that they are not lazy :)

**Comment by garf** [comment]
Done! Should not cause deprecation now.

**Pull request merged**